src/main/java/org/sefglobal/scholarx/service/ProgramService.java

```diff
@@ -378,4 +378,42 @@ public Mentor getSelectedMentor(long programId, long profileId)
         log.error(msg);
         throw new NoContentException(msg);
     }
+
+    /**
+     * Update the application of a {@link Mentee}
+     *
+     * @param profileId which is the Profile id of the {@link Mentee} to be updated
+     * @param programId  which is the Program id of the {@link Mentee} to be updated
+     * @param mentee    with the application of the mentee to be updated
+     * @return the updated {@link Mentee}
+     *
+     * @throws ResourceNotFoundException is thrown if the {@link Mentee} doesn't exist
+     * @throws BadRequestException       if the {@link Mentee} is not in the valid state
+     */
+    public Mentee updateMenteeData(long profileId, long programId, Mentee mentee)
+            throws ResourceNotFoundException, BadRequestException {
+        Optional<Mentee> optionalMentee = menteeRepository.findByProgramIdAndProfileId(profileId, programId);
+        if (!optionalMentee.isPresent()) {
+            String msg = "Error, Mentee by profile id: " + profileId + " and " +
+                    "mentor id: " + programId + " cannot be updated. " +
+                    "Mentee doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        }
+
+        Mentee existingMentee = optionalMentee.get();
+        if (EnrolmentState.PENDING.equals(existingMentee.getState())) {
+            existingMentee.setCourse(mentee.getCourse());
```

Comment:
Need to set the appliedMentor
```suggestion
            existingMentee.setAppliedMentor(mentee.AppliedMentor());
            existingMentee.setCourse(mentee.getCourse());
```

Reply:
@Piumal1999  I think this should be `getAppliedMentor()`. Is it?

Reply:
yeah